src/server/middleware/cosign/cosign.go

```diff
@@ -111,8 +111,8 @@ func CosignSignatureMiddleware() func(http.Handler) http.Handler {
 				Digest:        desc.Digest.String(),
 				Type:          model.TypeCosignSignature,
 			})
-			if err != nil {
-				logger.Errorf("failed to get cosign signature artifact: %s, error: %v", desc.Digest.String(), err)
+			if err != nil && !errors.IsConflictErr(err) {
+				logger.Errorf("failed to create cosign signature artifact: %s, error: %v", desc.Digest.String(), err)
 				return err
 			}
 		}
```

Comment:
Please wrap these lines in the tx, otherwise, the conflict error will break the DB conn if the conn is from a tx.

```
			_, err = accessory.Mgr.Create(ctx, model.AccessoryData{
				ArtifactID:    art.ID,
				SubArtifactID: subjectArt.ID,
				Size:          desc.Size,
				Digest:        desc.Digest.String(),
				Type:          model.TypeCosignSignature,
			})
			if err != nil && !errors.IsConflictErr(err) {
				logger.Errorf("failed to create cosign signature artifact: %s, error: %v", desc.Digest.String(), err)
				return err
			}
```